OvmfPkg/PlatformPei/Platform.c

```diff
@@ -470,6 +528,20 @@ BootModeInitialization (
 
   Status = PeiServicesInstallPpi (mPpiBootMode);
   ASSERT_EFI_ERROR (Status);
+
+  //
+  // Set BootMode based on FV header state
+  //
+  Status = ValidateFvHeader (FwVolHeader);

```

Comment:
This should probably be put before line `if (CmosRead8 (0xF) == 0xFE) {` and use the variable mBootMode:

```C
  //
  // Set BootMode based on FV header state
  //
  Status = ValidateFvHeader (FwVolHeader);

  if (EFI_ERROR (Status)) {
    mBootMode = BOOT_WITH_DEFAULT_SETTINGS;
    DEBUG ((DEBUG_INFO, "BootMode: Boot with default settings\n"));
  } else {
    mBootMode = BOOT_ASSUMING_NO_CONFIGURATION_CHANGES;
    DEBUG ((DEBUG_INFO, "BootMode: Boot boot assuming no configuration changes\n"));
```